tests/test_verifier.py

```
@@ -73,4 +74,29 @@ def test_verifier(self, m_resolver, m_deliver):
+    @patch.object(Verifier, '_can_deliver', side_effect=smtplib.SMTPHeloError(550, b'Blocked'))
+    @patch('dns.resolver.query', return_value=dns_response)
+    def test_verifier_returns_dict_on_smtp_helo_error(self, m_resolver, m_deliver):
+        # When the IP is blacklisted the server may reject the HELO command,
+        # raising SMTPHeloError.  verify() must not propagate the exception.
+        result = self.verifier.verify('user@example.com')
+        self.assertIsInstance(result, dict)
+        self.assertIn('message', result)
+        self.assertFalse(result['deliverable'])
+
+    @patch('dns.resolver.query', return_value=dns_response)
+    def test_can_deliver_initialises_deliverable_for_non_250_success(self, m_resolver):
```

Ruff (0.15.2)

[warning] Unused method argument: `m_resolver`

(ARG002)